sdk/typescript/src/builder/TransactionBlock.ts

```diff
@@ -624,16 +624,22 @@ export class TransactionBlock {
           provider,
         );
       }
-
       if (!this.blockData.gasConfig.budget) {
         const dryRunResult = await expectProvider(
           provider,
         ).dryRunTransactionBlock({
           transactionBlock: this.#blockData.build({
-            overrides: { gasConfig: { budget: String(MAX_GAS), payment: [] } },
+            overrides: {
+              gasConfig: {
+                budget: String(MAX_GAS),
+                // use user provided payment if it exists.
+                payment: this.#blockData.gasConfig.payment
+                  ? this.#blockData.gasConfig.payment
+                  : [],
+              },
```

Comment:
Can you revert this change? Actually, I don't think we should override the payment object because we use `MAX_GAS` on line 634.